src/pyhf/infer/calculators.py

```diff
@@ -0,0 +1,155 @@
+"""
+Calculators for Hypothesis Testing.
+
+The role of the calculators is to compute test statistic and
+provide distributions of said test statistic under various 
+hypotheses.
+
+Using the calculators hypothesis tests can then be performed.
+"""
+from .mle import fixed_poi_fit
+from .. import get_backend
+from .test_statistics import qmu
+
+
+def generate_asimov_data(asimov_mu, data, pdf, init_pars, par_bounds):
+    """Compute Asimov Dataset (expected yields at best-fit values) for a given POI value."""
+    bestfit_nuisance_asimov = fixed_poi_fit(asimov_mu, data, pdf, init_pars, par_bounds)
+    return pdf.expected_data(bestfit_nuisance_asimov)
+
+
+class AsymptoticTestStatDistribution(object):
+    """
+    The distribution the test statistic in the asymptotic case.
+
+    Note: These distributions are in -µ^/sigma space. In the ROOT
+    implementation the same sigma is assumed for both hypotheses
+    and p-values etc are computed in that space. This assumption
+    is necessarily valid, but we keep this for compatibility
+    reasons.
+
+    In the -µ^/sigma space, the test statistic (i.e. µ^/sigma) is
+    normally distributed with unit variance and its mean at 
+    the -µ', where µ' is the true poi value of the hypothesis.
+    """
+
+    def __init__(self, shift):
+        """
+        Asymptotic test statistic distribution.
+
+        Args:
+            shift: the displacement of the test statistic distribution
+
+        Returns:
+            distribution
+        
+        """
+        self.shift = shift
+
+    def pvalue(self, value):
+        """
+        Compute the p-value for a given value of the test statistic.
+        
+        Args:
+            value: the test statistic value.
+
+        Returns;
```

Comment:
This typo shows up a few places.

```suggestion
        Returns:
```